rustsat/src/types.rs

```diff
@@ -697,6 +704,56 @@ impl Assignment {
             Some(var![self.assignment.len() as u32 - 1])
         }
     }
+
+    /// Read a solution from a SAT solver's output given the path
+    pub fn from_solver_output_path<P: AsRef<Path>>(path: P) -> anyhow::Result<SolverOutput> {
+        let reader = std::io::BufReader::new(
+            instances::fio::open_compressed_uncompressed_read(path)
+                .context("failed to open reader")?,
+        );
+        instances::fio::parse_sat_solver_output(reader)
+    }
+
+    /// Given a value line of a SAT solver's solution create an Assignment with
+    /// this line's literal values already included
+    pub fn from_vline(line: &str) -> anyhow::Result<Self> {
+        let mut assignment = Assignment::default();
+        assignment.extend_from_vline(line)?;
+        Ok(assignment)
+    }
+
+    /// Parses and saves literals from a value line.
+    pub fn extend_from_vline(&mut self, line: &str) -> anyhow::Result<()> {
+        if !line.starts_with("v ") {
+            if line.is_empty() {
+                anyhow::bail!(InvalidVLine::Emptyline);
+            }
+
+            anyhow::bail!(InvalidVLine::InvalidTag(line.chars().next().unwrap()));
+        }
+
+        let line = &line[1..].trim_start();
+        for number in line.trim().split(' ').filter(|x| !x.is_empty()) {
```

Comment:
```suggestion
        let line = &line[1..];
        for number in line.split_whitespace() {
```
`split_whitespace` automatically removes empty elements and basically performs the trim operation.